core/query/src/main/java/datawave/core/query/configuration/QueryData.java

```diff
@@ -0,0 +1,576 @@
+     */
+    public QueryData(QueryData other) {
+        this.tableName = other.tableName;
+        this.query = other.query;
+        this.ranges = new HashSet<>(other.ranges);
+        this.columnFamilies = new HashSet<>(other.columnFamilies);
+        this.settings = new ArrayList<>(other.settings);
+        this.hashCode = other.hashCode;
+        this.rebuildHashCode = other.rebuildHashCode;
+        this.lastResult = other.lastResult;
+        this.finished = other.finished;
+    }
+
+    @Deprecated(since = "6.5.0", forRemoval = true)
+    public QueryData(String tableName, String query, Collection<Range> ranges, List<IteratorSetting> settings) {
+        setTableName(tableName);
+        setQuery(query);
+        setRanges(ranges);
+        setSettings(settings);
+    }
+
+    /**
+     * Weak copy constructor that updates the ranges
+     *
+     * @param other
+     *            another QueryData
+     * @param ranges
+     *            a collection of updated ranges
+     * @deprecated
+     */
+    @Deprecated(since = "6.5.0", forRemoval = true)
+    public QueryData(QueryData other, Collection<Range> ranges) {
+        this(other);
+        setRanges(ranges);
+    }
+
+    @Deprecated(since = "6.5.0", forRemoval = true)
+    public QueryData(String tableName, String queryString, List<Range> ranges, List<IteratorSetting> settings, Collection<String> columnFamilies) {
+        this(tableName, queryString, ranges, settings);
+        this.columnFamilies.addAll(columnFamilies);
+    }
+
+    // builder style methods
+
+    public QueryData withTableName(String tableName) {
+        this.tableName = tableName;
+        resetHashCode();
+        return this;
+    }
+
+    public QueryData withQuery(String query) {
+        this.query = query;
+        resetHashCode();
+        return this;
+    }
+
+    public QueryData withRanges(Collection<Range> ranges) {
+        this.ranges = ranges;
+        resetHashCode();
+        return this;
+    }
+
+    public QueryData withColumnFamilies(Collection<String> columnFamilies) {
+        this.columnFamilies = columnFamilies;
+        resetHashCode();
+        return this;
+    }
+
+    public QueryData withSettings(List<IteratorSetting> settings) {
+        this.settings = settings;
+        resetHashCode();
+        return this;
+    }
+
+    public void setSettings(List<IteratorSetting> settings) {
+        this.settings.clear();
+        if (settings != null) {
+            this.settings.addAll(settings);
+        }
+        resetHashCode();
+    }
+
+    public List<IteratorSetting> getSettings() {
+        return settings;
+    }
+
+    public void setQuery(String query) {
+        this.query = query;
+        resetHashCode();
+    }
+
+    public String getQuery() {
+        return query;
+    }
+
+    public String getTableName() {
+        return tableName;
+    }
+
+    public void setTableName(String tableName) {
+        this.tableName = tableName;
```

Comment:
resetHashCode()

Reply:
this `resetHashCode()` stuff feels really hokey, but ok.